src/pybind/mgr/dashboard/controllers/multi_cluster.py

```diff
@@ -205,13 +213,22 @@ def reconnect_cluster(self, url: str, username=None, password=None,
 
             cluster_token = self.check_cluster_connection(url, payload, username,
                                                           ssl_verify, ssl_certificate)
+            
+            prometheus_url = self._proxy('GET', url, 'api/multi-cluster/get_prometheus_api_url',
+                                         token=cluster_token)
+            
+            prometheus_access_info = self._proxy('GET', url, 'ui-api/multi-cluster/get_prometheus_access_info', token=cluster_token)
 
-        if username and cluster_token:
+        if username and cluster_token and prometheus_url and prometheus_access_info:
             if "config" in multicluster_config:
                 for _, cluster_details in multicluster_config["config"].items():
                     for cluster in cluster_details:
                         if cluster["url"] == url and cluster["user"] == username:
                             cluster['token'] = cluster_token
+                            cluster['prometheus_access_info'] = prometheus_access_info
+                            _remove_prometheus_targets(cluster['prometheus_url'])
+                            time.sleep(5)
```

Comment:
why do we need this sleep here? isn't the call to ` _remove_prometheus_targets` synchronous?

Reply:
Since _remove_prometheus_target method involves re-deploying the prometheus service. So I added a sleep here for 5s so the service can get up in that time and then we proceed on _set_prometheus_target method

Reply:
I agree. We can manually check if the daemon is running state if thats what we want. Probably thats more accurate